pkg/controllers/management/node/controller.go

```diff
@@ -717,3 +742,81 @@ func (m *Lifecycle) updateRawConfigFromCredential(data map[string]interface{}, r
 	}
 	return nil
 }
+
+func (m *Lifecycle) deleteV1Node(node *v3.Node) (runtime.Object, error) {
+	logrus.Debugf("Deleting v1.node for [%v] node", node.Status.NodeName)
+	if nodehelper.IgnoreNode(node.Status.NodeName, node.Status.NodeLabels) {
+		logrus.Debugf("Skipping v1.node removal for [%v] node", node.Status.NodeName)
+		return node, nil
+	}
+
+	if node.Status.NodeName == "" {
+		return node, nil
+	}
+
+	// get user context k8s client to delete downstream cluster node
+	cluster, err := m.clusterLister.Get("", node.Namespace)
+	if err != nil {
+		return node, err
+	}
+
+	userClient, err := m.clusterManager.UserContext(cluster.Name)
+	if err != nil {
+		return node, err
+	}
+
+	err = userClient.K8sClient.CoreV1().Nodes().Delete(
+		context.TODO(), node.Status.NodeName, metav1.DeleteOptions{})
+	if !kerror.IsNotFound(err) {
+		return node, err
+	}
+
+	return node, nil
+}
+
+func (m *Lifecycle) drainNode(node *v3.Node) error {
+	cluster, err := m.clusterLister.Get("", node.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if !nodehelper.DrainBeforeDelete(node, cluster) {
+		return nil
+	}
+
+	logrus.Infof("node [%s] requires draining before delete, attempting to drain", node.Spec.RequestedHostname)
+	kubeConfig, err := m.getKubeConfig(cluster)
+	if err != nil {
+		return fmt.Errorf("node [%s] error getting kubeConfig", node.Spec.RequestedHostname)
+	}
+
+	if node.Spec.NodeDrainInput == nil {
+		logrus.Debugf("node [%s] has no NodeDrainInput, creating one with 60s timeout",
+			node.Spec.RequestedHostname)
+		node.Spec.NodeDrainInput = &v3.NodeDrainInput{
+			Force:           false,
+			DeleteLocalData: false,
+			GracePeriod:     60,
+			Timeout:         60,
+		}
```

Comment:
Node object needs to be updated and should be using a copy.

Reply:
I can see why you might not need to update the node for this to work, perhaps you shouldn't assign it to the obj if it's not used anywhere else. Something like `nodeDrainInput = node.Spec.NodeDrainInput; if nodeDrainInput == nil ...`

Reply:
this is just a niceity, it should be set but incase it's not i quickly set it... after this the node gets deleted so there's no need to persist

Reply:
oh and it needs to be on the Node for GetDrainFlags to work, was just easier to put there as other functions in nodesyncer had to change to move that to pkg/node properly.

Reply:
and there are no other finalizers and you don't think other will be added? because this could mess up the cache.

Reply:
if it was nil before it can stay nil, it's not a problem. i just need it set quickly to run the drain.